ui/apps/platform/src/Components/DeploymentPortConfig.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+import {
+    Card,
+    CardBody,
+    DescriptionList,
+    DescriptionListDescription,
+    DescriptionListGroup,
+    DescriptionListTerm,
+    ExpandableSection,
+    Stack,
+    StackItem,
+} from '@patternfly/react-core';
+
+import { PortConfig } from 'types/deployment.proto';
+
+type DeploymentPortConfigProps = {
+    port: PortConfig;
+};
+
+function DeploymentPortConfig({ port }: DeploymentPortConfigProps) {
+    const [isExpanded, setIsExpanded] = useState(false);
+
+    const onToggle = (_isExpanded: boolean) => {
+        setIsExpanded(_isExpanded);
+    };
+
+    return (
+        <ExpandableSection
+            toggleText={port.name}
+            onToggle={onToggle}
+            isExpanded={isExpanded}
+            displaySize="large"
+            isWidthLimited
+        >
+            <Stack hasGutter>
+                <StackItem>
+                    <DescriptionList columnModifier={{ default: '2Col' }} isCompact>
+                        <DescriptionListGroup>
+                            <DescriptionListTerm>Container port</DescriptionListTerm>
+                            <DescriptionListDescription>
+                                {port.containerPort}
+                            </DescriptionListDescription>
+                        </DescriptionListGroup>
```

Comment:
oh i just remembered i made a component called `DescriptionListItem.tsx` that basically wraps the term and description in the group into one component if you wanted to simplify the code here a bit